mopidy_iris/system.py

```diff
@@ -20,49 +59,50 @@ def run(self):
 
         try:
             self.can_run()
-        except Exception as e:
+        except IrisSystemError as e:
             logger.error(e)
 
             error = {
-                'message': "Permission denied",
-                'description': str(e)
+                'message': e.reason,
+                'description': e.message
             }
             
             return {
                 'error': error
             }
 
-        logger.debug("sudo "+ self.path +"/system.sh "+ self.action)
-
-        proc = subprocess.Popen(["sudo", self.path+"/system.sh", self.action], 
-            stdout=subprocess.PIPE, 
-            stderr=subprocess.STDOUT)
+        command = self.get_command()
+        logger.debug("Running '%s'", os.fsdecode(b' '.join(command)))
+        proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         
         stdout,stderr = proc.communicate()
 
         if stderr:
-            logger.error(stderr.decode())
-            self.callback(None, { 'error': stderr.decode() })
+            error_string = os.fsdecode(stderr)
+            logger.error(error_string)
+            self.callback(None, {'error': error_string})
         else:
-            logger.info(stdout.decode())
-            self.callback({ 'output': stdout.decode() }, None)
-
-
+            response_string = os.fsdecode(stdout)
+            logger.info(response_string)
+            self.callback({'output': response_string}, None)
 
     ##
     # Check if we have access to the system script (system.sh)
     #
     # @return boolean or exception
     ##
     def can_run(self, *args, **kwargs):
+        if not self.script_path.is_file():

```

Comment:
I was thinking to get rid of this check I did, seems pointless to check a non-runtime issue here.